lib/rmt/mirror/base.rb

```diff
@@ -130,18 +131,26 @@ def need_to_download?(ref)
     true
   end
 
-  def move_directory(source:, destination:)
-    FileUtils.mv(source, destination, force: true)
-    FileUtils.chmod(0o755, destination)
+  def cleanup_stale_metadata
+    # A bug introduced in 2.16 writes metadata into its own directory if exists having
+    # directory structure like repodata/repodata.
+    # see: https://github.com/SUSE/rmt/issues/1136
+    FileUtils.remove_entry(repository_path('repodata', 'repodata')) if Dir.exist?(repository_path('repodata', 'repodata'))
+
+    # With 1.0.0 a backup mechanism was introduced creating .old_* backups of metadata which was never really used
+    # we remove these files now from the mirrored repositories
+    # see: https://github.com/SUSE/rmt/pull/1120/files#diff-69bc4fdeb7aa7ceab24bec11c65a184357e5b71317125516edfa2d819653a969L131
```

Comment:
Could we maybe leave a note about the `.old_` and `.backup_`  thing in the PR? 
I mean about how it was earlier `.old_` but then we changed it to `.backup_` but since it we also removed backups as a part of the 2.16 release etc etc.

Reply:
I added an additional note to describe this!